common/configuration/obfuscator.go

```diff
@@ -0,0 +1,97 @@
+package configuration
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/TBD54566975/ftl/internal/slices"
+)
+
+type ObfuscatorProvider interface {
+	obfuscator() Obfuscator
+}
+
+// Obfuscator hides and reveals a value, but does not provide real security
+// instead the aim of this Obfuscator is to make values not easily human readable while attaching a comment
+//
+// Obfuscation is done by XOR-ing the input with the AES key. Length of key must be 16, 24 or 32 bytes (corresponding to AES-128, AES-192 or AES-256 keys).
+//
+// Example obfuscation result:
+// # Comments appear at top
+// # Multiple lines are supported
+// d144b654d69a438cf7bcaaa59dee430e51d5c3cbeb6f61d8bc3f79f3484e7234fab5280ac57678d68e6c
+type Obfuscator struct {
+	key []byte
+}
+
+// Obfuscate takes a value and returns an obfuscated value (encoded in hex) with a comment
+func (o Obfuscator) Obfuscate(input []byte, comment string) ([]byte, error) {
+	encoded, err := o.encode(input)
+	if err != nil {
+		return nil, err
+	}
+	// build output by prepending comments
+	lines := []string{}
+	for _, line := range strings.Split(comment, "\n") {
+		lines = append(lines, "# "+line)
+	}
+	lines = append(lines, hex.EncodeToString(encoded))
+	return []byte(strings.Join(lines, "\n")), nil
+}
+
+// Reveal takes an obfuscated value, ignores any comments (lines starting with '#') and de-obfuscates the hex encoded value
+func (o Obfuscator) Reveal(input []byte) ([]byte, error) {
```

Comment:
Maybe let's be backwards compatible, as it won't be that much work and will avoid existing secrets having to be migrated.

Something like checking if `strings.ContainsRune("0123456789abcdefABCDEF", hexEncoded[0])`

Reply:
Though, related, perhaps we should use base64 instead of hex, as it encodes quite a bit smaller.

Reply:
Switched to base64 and made it backwards compatible. It won't catch cases where the secret was just a number but still better than nothing. Ran it with the inline toml in our partner project and it all worked went smoothly.